### PR TITLE
fixes #884

### LINKED ISSUE #884: Error :  Function calls are not supported in decorators but 'QuillModule' was called.
I get this error only when I run a Production build, but development build works just fine. I am not sure if the problem is with AOT compile. My Angular version is 8.

I have the following entry in  my **app,modue.ts**

import { QuillModule } from 'ngx-quill';

imports: [
....
    QuillModule.forRoot(),
]

Entries in package.json
===============
"ngx-quill": "^9.1.1",
 "quill": "^1.3.7",
 "quill-mention": "^2.2.5",

Error Below
===============
image

really appreciate your help with this.
Thanks

### PR DESCRIPTION
**Pull request by jpike88**
fixes #884

### CODE REVIEW

**Comment by KillerCodeMonkey** [comment]
but now the config itself is not injectable ;-). because it is not present. But i do not get it why it failes anyways... that is the angular way to define a global config that can be overwritten.

There is nothing special in there. I only added the "ModuleWithProvider" to the forRoot method.

File: projects/ngx-quill/src/lib/quill-view.component.ts [hunk 1]
```diff
@@ -87,7 +87,7 @@ export class QuillViewComponent implements AfterViewInit, OnChanges {
 
     const Quill = await this.service.getQuill()
 
-    const modules = Object.assign({}, this.modules || (this.config.modules || defaultModules))
+    const modules = this.service.config.modules;
```
Comment (KillerCodeMonkey) [inline projects/ngx-quill/src/lib/quill-view.component.ts:90]: please remove the `;`

**Comment by jpike88** [comment]
It fails because ```forRoot``` must be deterministic, there can't be any transformation in that function anymore. It's an evil pain in the ass this one.

Relevant reading

https://github.com/angular/angular/issues/23609#issuecomment-561615821

**Comment by jpike88** [comment]
It makes sense to use the service as the configuration provider for components etc, this is probably better than the checks being done to resolve the modules object for quill

**Comment by KillerCodeMonkey** [comment]
> It fails because `forRoot` must be deterministic, there can't be any transformation in that function anymore. It's an evil pain in the ass this one.
> 
> Relevant reading (but will just make you sad and confused)
> 
> [angular/angular#23609 (comment)](https://github.com/angular/angular/issues/23609#issuecomment-561615821)

ultra lame!

**Comment by KillerCodeMonkey** [comment]
@jpike88 please run test cases and linting, so to avoid functionalities will break ;-)

**Comment by jpike88** [comment]
no problem I'm going to continue on until this is fixed now

**Comment by jpike88** [comment]
ugh I'm running into problems here... any tips

**Comment by KillerCodeMonkey** [comment]
@jpike88 no rush. i can check it this evening if i have the time.

**Comment by jpike88** [comment]
I did it

**Comment by KillerCodeMonkey** [comment]
awesome!

**Comment by jpike88** [comment]
great, looking forward to release, I'll confirm it works

**Comment by KillerCodeMonkey** [comment]
yep i test it against my demo repos and then i release a new version

**Comment by KillerCodeMonkey** [comment]
released 10.0.3. thanks again man :)

**Comment by jpike88** [comment]
confirmed working in my repo. no worries dude!

**Pull request merged**